agent/pulp/agent/gofer/pulpplugin.py

```diff
@@ -118,6 +151,35 @@ def get_secret():
     return bundle.uid()
 
 
+class Attach(Thread):
+    """
+    This thread (task) persistently:
+      - validates the registration status
+      - if registered, updates the plugin settings and attach.
+      - if not registered, detach the plugin.
+    The reason for doing this in a thread is that we don't
+    want to block in the initializer.
+    """
+    
```

Comment:
There is some extra whitespace here that will make flake8 complain, and shows up as red in `git diff HEAD^`. The extra whitespace is in a few other places as well.